File: 프로그래머스/unrated/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    answer = 0
    data = ["aya", "ye", "woo", "ma"]

    for b in babbling:
        check = -1

        while True:
            if b == "":
                answer += 1
                break

            flag = False
            for idx, value in enumerate(data):
                if check != idx and b.startswith(value):
                    check = idx
                    b = b[len(value):]
                    flag = True

            if not flag:
                break
    return answer

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.java
import java.util.*;

class Solution {
    static String[] speaks = new String[] { "aya", "ye", "woo", "ma" };

    public int solution(String[] babbling) {
        int cnt = 0;
        for (String b : babbling) {
            if(canSpeak(b)) cnt += 1;
        }
        
        return cnt;
    }
    
    public boolean canSpeak(String b) {
        String prevSpeak = "";
        while(b.length() > 0) {
            boolean flag = false;
            
            for (String s : speaks) {
                if(prevSpeak.equals(s)) continue;
                if(b.startsWith(s)) {
                    b = b.substring(s.length());
                    prevSpeak = s;
                    flag = true;
                }
            }
            
            if(!flag) return false;
        }
        
        return true;
    }
}